src/SumoCoders/FrameworkCoreBundle/Tests/Mail/MessageFactoryTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace SumoCoders\FrameworkCoreBundle\Tests\Mail;
+
+use SumoCoders\FrameworkCoreBundle\Mail\MessageFactory;
+
+class MessageFactoryTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var MessageFactory
+     */
+    protected $messageFactory;
+
+    protected function setUp()
+    {
+        $temp = tempnam(sys_get_temp_dir(), 'message_factory_test');
+        $this->messageFactory = new MessageFactory(
+            $this->getTemplating(),
+            'mails/default_email.html.twig',
+            $temp
+        );
+    }
+
+    /**
+     * @inherit
+     */
+    protected function tearDown()
+    {
+        $this->messageFactory = null;
+    }
+
+    /**
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    protected function getTemplating()
+    {
+        $templating = $this->getMock('\Symfony\Bundle\FrameworkBundle\Templating\EngineInterface');
+        $templating
+            ->method('render')
+            ->willReturn('<html><head></head><body><p>And I, le content</p></body></html>');
+
+        return $templating;
+    }
+
+    public function testIfTheDefaultsAreEmptyWhenNotSet()
```

Comment:
testIf is kind of an anti pattern in test naming. You should be able to read the test name as a sentence if you strip 'test' from it. But "If the defaults are empty when nog set" is not a great sentence.